verifier/source_reader_service.go

```diff
@@ -102,57 +104,42 @@ func NewSourceReaderService(
 
 // Start begins reading messages and pushing them to the messages channel.
 func (r *SourceReaderService) Start(ctx context.Context) error {
-	r.mu.Lock()
-	defer r.mu.Unlock()
-
-	if r.isRunning {
-		return nil // Already running
-	}
-
-	r.logger.Infow("Starting SourceReaderService",
-		"chainSelector", r.chainSelector,
-		"topic", r.ccipMessageSentTopic)
-
-	// Test connectivity before starting
-	if err := r.testConnectivity(ctx); err != nil {
-		r.logger.Errorw("Connectivity test failed", "error", err)
-		return err
-	}
-
-	r.isRunning = true
-	r.wg.Add(1)
+	return r.StartOnce("SourceReaderService", func() error {
+		r.logger.Infow("Starting SourceReaderService",
+			"chainSelector", r.chainSelector,
+			"topic", r.ccipMessageSentTopic)
+
+		// Test connectivity before starting
+		if err := r.testConnectivity(ctx); err != nil {
+			r.logger.Errorw("Connectivity test failed", "error", err)
+			return err
+		}
 
-	go r.eventMonitoringLoop(ctx)
+		r.wg.Add(1)
+		go r.eventMonitoringLoop(ctx)
 
-	r.logger.Infow("SourceReaderService started successfully")
-	return nil
+		r.logger.Infow("SourceReaderService started successfully")
+		return nil
+	})
 }
 
 // Stop stops the reader and closes the messages channel.
 func (r *SourceReaderService) Stop() error {
-	r.mu.Lock()
-	if !r.isRunning {
-		r.mu.Unlock()
-		return nil // Already stopped
-	}
+	return r.StopOnce("SourceReaderService", func() error {
+		r.logger.Infow("Stopping SourceReaderService")
 
-	r.logger.Infow("Stopping SourceReaderService")
+		close(r.stopCh)
```

Comment:
Maybe stopCh is not needed if we're going to use a context cancel to stop the loop?

Reply:
Yeah, I want to dive deeper there in next PR